assets/js/settings/shared/set-setting.js

```diff
@@ -3,17 +3,35 @@
  */
 import { allSettings } from './settings-init';
 
+/**
+ * External dependencies
+ */
+import deprecated from '@wordpress/deprecated';
+
 /**
  * Sets a value to a property on the settings state.
  *
  * @export
  * @param {string}   name                        The setting property key for the
  *                                               setting being mutated.
- * @param {*}    value                       The value to set.
+ * @param {*}        value                       The value to set.
  * @param {Function} [filter=( val ) => val]     Allows for providing a callback
  *                                               to sanitize the setting (eg.
  *                                               ensure it's a number)
+ *
+ * @todo  Remove setSetting function from `@woocommerce/settings`.
+ *
+ *  The `wc.wcSettings.setSetting` function was deprecated beginning with WooCommerce Blocks 3.2.0
+ *  and can be removed completely with WooCommerce Blocks 3.8.0 (that gives 6 versions of the
+ *  feature plugin for warning and 3 versions of WooCommerce core).
  */
 export function setSetting( name, value, filter = ( val ) => val ) {
+	deprecated( 'setSetting', {
+		version: '3.8.0',
+		alternative: `a locally scoped value for "${ name }" instead`,
```

Comment:
```suggestion
		alternative: `a locally scoped value for "${ name }"`,
```
this causes a double instead.